src/logion/services/extrinsicsbuilder.ts

```diff
@@ -23,13 +23,21 @@ export class ExtrinsicsBuilder {
     async build(): Promise<JsonExtrinsic[]> {
         const extrinsics = this.block.extrinsics;
         const builders: ExtrinsicBuilder[] = new Array(extrinsics.length);
+
+        for (let i = 0; i < extrinsics.length; i++) {
+            const extrinsicBuilder = this.createBuilder(extrinsics[i]);
+            if (extrinsicBuilder) {
+                builders[i] = extrinsicBuilder;
+            }
+        }
```

Comment:
This is not equivalent to previous code but not sure if there is an actual impact. Let's discuss to make sure.

Reply:
To remain on the safe side, I have adapted the code to keep the previous logic while still fixing the bug with the missing events.